integration/query_frontend_test.go

```diff
@@ -233,10 +255,14 @@ func runQueryFrontendTest(t *testing.T, cfg queryFrontendTestConfig) {
 	require.NoError(t, queryFrontend.WaitSumMetrics(e2e.Equals(1), "thanos_memcached_dns_provider_results"))
 	require.NoError(t, queryFrontend.WaitSumMetrics(e2e.Greater(0), "thanos_memcached_dns_lookups_total"))
 
-	// Wait until both the distributor and querier have updated the ring.
-	// The distributor should have 512 tokens for the ingester ring and 1 for the distributor ring
-	require.NoError(t, distributor.WaitSumMetrics(e2e.Equals(512+1), "cortex_ring_tokens_total"))
-	require.NoError(t, querier.WaitSumMetrics(e2e.Equals(512), "cortex_ring_tokens_total"))
+	// Wait until distributor and querier have updated the ingesters ring.
+	require.NoError(t, distributor.WaitSumMetricsWithOptions(e2e.Equals(1), []string{"cortex_ring_members"}, e2e.WithLabelMatchers(
```

Comment:
Question: we don't need to wait for "query-scheduler" ring because both query-frontends and queriers only report "ready" after they connect to query-scheduler. Is that right?

Reply:
The query-frontend report "ready" after it connected to at least 1 query-scheduler, so the query will be successfully enqueued. The querier doesn't need to do such check, because as soon as successfully connect to the query-scheduler, it will pick up the query, run it and send the response back to the query-frontend.

Reply:
Thanks!